src/components/topic/TopicDirective.js

```diff
@@ -8,14 +8,30 @@ goog.require('ga_topic_service');
   ]);
 
   module.directive('gaTopic',
-      function($rootScope, gaTopic) {
+      function($rootScope, $translate, $q, gaTopic) {
         return {
           restrict: 'A',
           templateUrl: 'components/topic/partials/topic.html',
           scope: {},
           link: function(scope, element, attrs) {
             var modal = element.find('.modal');
 
+            var translationsReady = function() {
+              var deferred = $q.defer();
+              $translate.onReady(function() {
```

Comment:
`$translate.onReady` already returns a promise, you can use it directly in the `$q.all`